test/e2e/helpers.go

```diff
@@ -75,7 +75,7 @@ const (
 	retryableOperationTimeout             = 30 * time.Second
 	retryableDeleteOperationTimeout       = 3 * time.Minute
 	retryableOperationSleepBetweenRetries = 3 * time.Second
-	helmInstallTimeout                    = 3 * time.Minute
+	helmInstallTimeout                    = 10 * time.Minute
```

Comment:
I assume we're increasing this timeout in response to recent flakes? Did you see this comment here?

- https://kubernetes.slack.com/archives/CEX9HENG7/p1677633345332789?thread_ts=1677618484.842959&cid=CEX9HENG7

I'm not sure we should be relying upon this much tolerance for helm install failures, maybe we're doing the helm installs too early?

Reply:
/hold

I think you're right, let me try switching the order

Reply:
/hold cancel

Switching the order won't work because the CAPI framework func checks for a node ref on the Machine which won't be there until cloud-provider is installed, but I added a check before installing the helm chart to make sure the api server is reachable and reverted the timeout increase.